src/expressions/dataTypes/Value.ts

```diff
@@ -5,67 +5,390 @@ export default class Value {
+ * @public
+ */
+export type ParameterType = SequenceType | EllipsisType;
+
+/**
+ * The type of a sequence, it contains the ValueType and the multiplicity of the sequence.
+ * @public
+ */
+export type SequenceType = {
+	mult: SequenceMultiplicity;
+	type: ValueType;
+};
+
+/**
+ * A function to retrieve wether a type starts with xs.
+ * @param inType input type.
+ * @returns a boolean, true when the type starts with XS
+ */
+export function startWithXS(inType: ValueType): boolean {
+	return startWithXSLookupTable[inType];
+}
+
+const startWithXSLookupTable = new Map([
+	[ValueType.XSBOOLEAN, true],
+	[ValueType.XSSTRING, true],
+	[ValueType.XSNUMERIC, true],
+	[ValueType.XSDOUBLE, true],
+	[ValueType.XSDECIMAL, true],
+	[ValueType.XSINTEGER, true],
+	[ValueType.XSFLOAT, true],
+	[ValueType.XSDATE, true],
+	[ValueType.XSTIME, true],
+	[ValueType.XSDATETIME, true],
+	[ValueType.XSDATETIMESTAMP, true],
+	[ValueType.XSGYEARMONTH, true],
+	[ValueType.XSGYEAR, true],
+	[ValueType.XSGMONTHDAY, true],
+	[ValueType.XSGMONTH, true],
+	[ValueType.XSGDAY, true],
+	[ValueType.XSYEARMONTHDURATION, true],
+	[ValueType.XSDAYTIMEDURATION, true],
+	[ValueType.XSDURATION, true],
+	[ValueType.XSUNTYPEDATOMIC, true],
+	[ValueType.XSANYURI, true],
+	[ValueType.XSBASE64BINARY, true],
+	[ValueType.XSHEXBINARY, true],
+	[ValueType.XSQNAME, true],
+	[ValueType.XSNCNAME, true],
+	[ValueType.XSNAME, true],
+	[ValueType.XSENTITY, true],
+	[ValueType.XSNONPOSITIVEINTEGER, true],
+	[ValueType.XSNEGATIVEINTEGER, true],
+	[ValueType.XSPOSITIVEINTEGER, true],
+	[ValueType.XSNONNEGATIVEINTEGER, true],
+	[ValueType.XSLONG, true],
+	[ValueType.XSINT, true],
+	[ValueType.XSSHORT, true],
+	[ValueType.XSBYTE, true],
+	[ValueType.XSUNSIGNEDINT, true],
+	[ValueType.XSUNSIGNEDLONG, true],
+	[ValueType.XSUNSIGNEDBYTE, true],
+	[ValueType.XSUNSIGNEDSHORT, true],
+	[ValueType.XSERROR, true],
+	[ValueType.XSENTITIES, true],
+	[ValueType.XSIDREF, true],
+	[ValueType.XSID, true],
+	[ValueType.XSIDREFS, true],
+	[ValueType.XSNOTATION, true],
+	[ValueType.XSANYSIMPLETYPE, true],
+	[ValueType.XSANYATOMICTYPE, true],
+	[ValueType.ATTRIBUTE, false],
+	[ValueType.XSNORMALIZEDSTRING, true],
+	[ValueType.XSNMTOKENS, true],
+	[ValueType.XSNMTOKEN, true],
+	[ValueType.XSLANGUAGE, true],
+	[ValueType.XSTOKEN, true],
+	[ValueType.NODE, false],
+	[ValueType.ELEMENT, false],
+	[ValueType.DOCUMENTNODE, false],
+	[ValueType.TEXT, false],
+	[ValueType.PROCESSINGINSTRUCTION, false],
+	[ValueType.COMMENT, false],
+	[ValueType.ITEM, false],
+	[ValueType.FUNCTION, false],
+	[ValueType.MAP, false],
+	[ValueType.ARRAY, false],
+]);
+
+/**
+ * Creates a string value from a base type, generally follows the 'xs:type' notation, where
+ * xs stands for XML Schema
+ * @param input The base type to get the string of
+ * @returns A string corresponding to that base type.
+ */
+export function valueTypeToString(input: ValueType): string {
+	const typeToStringMap: Record<ValueType, string> = {
```

Comment:
you can hoist this map to the module level. right now it looks like it needs to be declared whenever the function starts